rust/private/rustc.bzl

```diff
@@ -1149,7 +1155,10 @@ def rustc_compile_action(
     rustc_output = crate_info_dict.get("rustc_output", None)
     rustc_rmeta_output = crate_info_dict.get("rustc_rmeta_output", None)
 
-    cc_toolchain, feature_configuration = find_cc_toolchain(ctx)
+    extra_disabled_features = [RUST_LINK_CC_FEATURE]
+    if crate_info_dict["type"] in ["bin", "cdylib"] and any([d.cc_info for d in crate_info_dict["deps"].to_list()]):
```

Comment:
This would only be true if a direct dependency is a C/C++ dependency; what's the intended behavior when there is a transitive C/C++ dependency?

Reply:
That's a good question. Looks like I can utilize the collected `dep_info` to include all the transitive deps as well. Updated the PR.